freqtrade/data/history.py

```diff
@@ -126,9 +142,15 @@ def load_pair_history(pair: str,
     :param exchange: Exchange object (needed when using "refresh_pairs")
     :param fill_up_missing: Fill missing values with "No action"-candles
     :param drop_incomplete: Drop last candle assuming it may be incomplete.
+    :param startup_candles: Additional candles to load at the start of the period
     :return: DataFrame with ohlcv data
     """
 
+    timerange_startup = deepcopy(timerange)
+    if startup_candles and timerange_startup:
+        logger.info('Using indicator startup period: %s ...', startup_candles)
+        timerange_startup.subtract_start(timeframe_to_seconds(ticker_interval) * startup_candles)
```

Comment:
calling `adjust_start_if_necessary()` here is wrong - this is before loading the data.

The duplication is not really avoidable (also, it's just one multiplication) ... 
I can move the multiplication to `subtract_start()` - however it'll still be duplicated since it's needed in both `subtract_start()` and `adjust_start_if_necessary` - and i don't intend to write a helper function for this calculation (since it would always need both variables anyway).
My thought was that using seconds in `subtract_start` may make it reusable in the future, where real seconds (not candles) need to be added or removed - but i can move the calculation there, no problem.

there's two parts to this PR ... (assuming 100 startupcandles):

* move loaded data timerange by 100 candles into the past - so 20190101 becomes 20181231 16:xx:xx - done by this section
  * then try to load this "extendet" period of data

* move requested timerange by 100 candles into the future if the above condition could not be fullfilled (if your data starts at 20190101). Adjusting that if necessary is what `adjust_start_if_necessary()` does.

Reply:
Thanks for explanation, let's leave it as it is here.
👍